PIPS/pip-2.md

```diff
@@ -0,0 +1,50 @@
+---
+pip: 2
+title: Relayer Fees
+status: Draft
+author: John Wong (@jwheur)
+created: 2020-06-30
+---
+
+## Simple Summary
+The current LockProxy implementation does not have an in-built method for relayers to charge fees for their work.
+
+This PIP proposes a simple method to charge fees.
+
+## Motivation
+Network fees can be very expensive. For example, the total Ethereum network fees for some popular DApps can cost up to 500k USD over 30 days (source: https://ethgasstation.info/).
+
+## Specification
```

Comment:
Point 1: 
I have some doubts about this implementation and the meaning of `feeAmount` field.
If the `feeAssetHash `is same as `TxArgs.fromAssetHash`, then we can deduct `feeAmount` from `TxArgs.amount` directly and update `TxArgs.amount` directly before `TxArgs` is serialized. The receiver at `toChainId` will receive updated `TxArgs.amount` value of asset `TxArgs.toAssetHash`.

If the `feeAssetHash` is same as `TxArgs.toAssetHash`, then at `toChainId` side after `TxArgs` is deserialized, the receiver at `toChainId` will receive updated `TxArgs.amount - feeAmount` value of asset `TxArgs.toAssetHash`.

Otherwise, at `toChainId` side, the receiver will receive `TxArgs.amount` value of asset `TxArgs.toAssetHash`.

In the above scenarios, the amount of `TxArgs.toAssetHash` the receiver will receive differs, and that may be confused. Recommend clarify that.

Point 2:
This proposal gives declaration about the interface of `lock` and `unlock`. It may be better to give the pseudocode in details, like, will this proposal change `TxArgs` struct of pip-1? If so, what should the new struct be defined as? How to deal with the new `TxArgs` in both `lock` and `unlock` method.

Reply:
I think Point 1 and Point 2 make a lot of sense, will update the description to make it clearer.

Reply:
if you want to transfer asset from one chain to another chain, you can pay some current asset as cross chain fee, then the Relayer will help you to transfer it.